js/improve onboarding

jskrzypek opened the pull request:
- feat: Use more portable home-dir
- feat: Add custom dir-prefix to install system
- fix: Disable dependency on removed external addon

A number of these changes may be superceded by the vendoring of the addons.

jskrzypek committed:
feat: Use new babashka.fs/home

For better portability and longevity, remove bb-godot.tasks/home-dir in favor of
the new babashka.fs/home.

jskrzypek committed:
feat: Add custom dir-prefix to install system

The Addons system may be in the process of being removed but as long as the
commands exist here, other users may want the repos cloned in a different
location than directly under their home directory. This change makes the
addons-* and install-script-templates commands take variadic additional args
from the command line, each word of which will be interpreted as a part of the
path under the user's $HOME directory in which they want the repos to be cloned,
or in the case of the script_templates, where the respective repo can be found
on the local machine.

russmatney commented on src/bb_godot/tasks.clj:
```diff
@@ -17,7 +17,7 @@
 (defn home-dir []
   (-> (bb.tasks/shell {:out :string}
-                      "zsh -c 'echo -n ~'")
+                      "sh -c 'printf %s $HOME'")
```
i think we can move to `babashka.fs/home` here - it's somewhat recent but should also be cross-platform and a bit cleaner

russmatney reviewed (approved):
Thanks for this, it definitely fixes a few baked-in assumptions (directory structure, for one). I think the dep handling impled here is important at a certain scale (working on multiple projects with multiple devs), but for now vendoring the deps simplifies usage for non-dev consumers. Still, thanks for making progress on this code - it may one day evolve into a stronger godot addon/package manager.

One detail is the that `babashka.fs` provides a `/home` function now, which might be preferred to the zsh/sh home-eval (tho it likely does the same thing, looking up `$HOME` in the env. But, no need to wait, i'll merge this as-is (with one conflict resolved in the github ui)

jskrzypek committed:
feat: Make the git status output for addons better

Rather than the full message for the git status output, which gets a bit lost
when being emitted for each addon repo, use git status -sb to show the short
format with the branch and tracking information.

Additionally, it seems prudent to fetch from the remote prior to checking the
status, to see how much divergence has taken place.
